sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/TryEval.scala

```diff
@@ -227,7 +228,8 @@ case class TryToBinary(
 
   def this(expr: Expression, formatExpression: Expression) =
     this(expr, Some(formatExpression),
-      TryEval(ToBinary(expr, Some(formatExpression), nullOnInvalidFormat = true)))
+      TryEval(ToBinary(expr, Some(TryToBinary.checkFormat(formatExpression)),
```

Comment:
Could you perform the check in `checkInputDataTypes()` as we do that in other expression, please.

BTW, exceptions from expressions constructors are wrapped by `AnalysisException` additionally which is not convenient to users.